Sources/PackagePlugin/Context.swift

```diff
@@ -39,14 +41,32 @@ public struct PluginContext {
     /// target on which the package plugin target depends. This function throws
     /// an error if the tool cannot be found. The lookup is case sensitive.
     public func tool(named name: String) throws -> Tool {
-        if let path = self.toolNamesToPaths[name] { return Tool(name: name, path: path) }
+        if let path = self.toolNamesToPaths[name] {
+            return Tool(name: name, path: path)
+        } else {
+            for dir in toolSearchDirectories {
+#if os(Windows)
+                let hostExecutableSuffix = ".exe"
+#else
+                let hostExecutableSuffix = ""
+#endif
+                let path = dir.appending(name + hostExecutableSuffix)
+                if FileManager.default.isExecutableFile(atPath: path.string) {
```

Comment:
should this be an addition / exertion on FileSystem instead hard coded to local one?